### PR TITLE
Hide admin notices on editor and dashboard screens

### PR DESCRIPTION
**Pull request by swissspidy**
Fixes this case:

Screenshot 2020-04-02 at 14 16 13

### CODE REVIEW

File: assets/src/dashboard/style.css [hunk 1]
```diff
@@ -26,7 +26,8 @@ body {
 }
 
 #screen-meta,
-#screen-meta-links {
+#screen-meta-links,
+#screen-meta-links ~ .notice {
```
Comment (spacedmonkey) [inline assets/src/dashboard/style.css]: Why not just `.notice`?

**Comment by spacedmonkey** [comment]
Why not just remove all actions on `admin_notices` on just these screens?

File: assets/src/dashboard/style.css (hunk 1 above)
Comment (swissspidy) [reply assets/src/dashboard/style.css]: Seemed too broad

**Comment by swissspidy** [comment]
I fear that there might be edge cases where that doesn't work due to different priorities etc.

Would probably require something like this:

```php
function remove() {
	// check for current screen, etc.

	remove_all_actions('admin_notices');
	remove_all_actions('all_admin_notices');
	add_action( 'admin_notices', 'remove', PHP_INT_MIN );
	add_action( 'all_admin_notices', 'remove', PHP_INT_MIN );
}

add_action( 'in_admin_header', 'remove', PHP_INT_MAX );
```

Which perhaps we can still do, but this CSS fix seems like a good quick win.

Or did you have something else in mind?

**Comment by spacedmonkey** [comment]
@swissspidy I didn't think about it being that hard, just thought it would be `remove_all_actions('admin_notices');`. If it is a pain, then I am okay with a CSS workaround for now. 

I had a question about css selector, but apart from that, I will approve.

File: assets/src/edit-story/style.css [hunk 2]
```diff
@@ -21,7 +21,8 @@
  */
 
 #screen-meta,
-#screen-meta-links {
+#screen-meta-links,
+#screen-meta-links ~ .notice {
```
Comment (barklund) [inline assets/src/edit-story/style.css]: I currently have an `.update_nag` on my screen:

Screen Shot 2020-04-02 at 11 59 04 AM

That wouldn't be caught by this, would it? And I feel it shouldn't go here - at least not in the editor!

HTML is:

```html
<div id="screen-meta-links">
  <div id="screen-options-link-wrap" class="hide-if-no-js screen-meta-toggle">
    <button type="button" id="show-settings-link" class="button show-settings" aria-controls="screen-options-wrap" aria-expanded="false">Screen Options</button>
  </div>
</div>
<div class="update-nag">
  <a href="https://wordpress.org/support/wordpress-version/version-5-4/">WordPress 5.4</a> is available! 
  <a href="http://one.wordpress.test/wp-admin/update-core.php" aria-label="Please update WordPress now">Please update now</a>.
</div>
```

Comment (swissspidy) [reply assets/src/edit-story/style.css]: Good catch! That should definitely be hidden as well.

Comment (swissspidy) [reply assets/src/edit-story/style.css]: (and you should definitely update 😄 )

**Comment by swissspidy** [comment]
@spacedmonkey @barklund Mind having another look? Note that I filed #997 for after this.

**Review by miina — APPROVED** [review]
Works well!

**Pull request merged**